guides/Getting Started.md

```diff
@@ -0,0 +1,696 @@
+  first_name: "Jane", id: 3, last_name: "Smith"}]
+```
+
+### Fetch a single record based on ID
+
+To fetch a record based on its ID, you use the `get` function:
+
+```elixir
+Friends.Person |> Friends.Repo.get(1)
+%Friends.Person{__meta__: #Ecto.Schema.Metadata<:loaded>, age: 28,
+ first_name: "Ryan", id: 1, last_name: "Bigg"}
+```
+
+### Fetch a single record based on a specific attribute
+
+If we want to get a record based on something other than the `id` attribute, we can use `get_by`:
+
+```elixir
+ Friends.Person |> Friends.Repo.get_by(first_name: "Ryan")
+ %Friends.Person{__meta__: #Ecto.Schema.Metadata<:loaded>, age: 28,
+  first_name: "Ryan", id: 1, last_name: "Bigg"}
+```
+
+### Filtering results
+
+If we want to get multiple records matching a specific attribute, we can use `where`:
+
+```elixir
+Friends.Person |> Ecto.Query.where(last_name: "Smith") |> Friends.Repo.all
+```
+
+```elixir
+[%Friends.Person{__meta__: #Ecto.Schema.Metadata<:loaded>, age: 27,
+  first_name: "John", id: 2, last_name: "Smith"},
+ %Friends.Person{__meta__: #Ecto.Schema.Metadata<:loaded>, age: 26,
+  first_name: "Jane", id: 3, last_name: "Smith"}]
+```
+
+If we leave off the `Friends.Repo.all` on the end of this, we will see the query Ecto generates:
+
+```
+#Ecto.Query<from p in Friends.Person, where: p.last_name == "Smith">
+```
+
+We can also use this query syntax to fetch these same records:
+
+```elixir
+Ecto.Query.from(p in Friends.Person, where: p.last_name == "Smith") |> Friends.Repo.all
+```
+
+One important thing to note with both query syntaxes is that they require variables to be pinned, using the pin operator (`^`). Otherwise, this happens:
+
+```elixir
+last_name = "Smith"
+Friends.Person |> Ecto.Query.where(last_name: last_name) |> Friends.Repo.all
+```
+
+```
+** (Ecto.Query.CompileError) variable `last_name` is not a valid query expression.
+  Variables need to be explicitly interpolated in queries with ^
+             expanding macro: Ecto.Query.where/2
+             iex:1: (file)
+    (elixir) expanding macro: Kernel.|>/2
+             iex:1: (file)
+```
+
+The same will happen in the longer query syntax too:
+
+```elixir
+Ecto.Query.from(p in Friends.Person, where: p.last_name == last_name) |> Friends.Repo.all
+```
+
+```
+** (Ecto.Query.CompileError) variable `last_name` is not a valid query expression. 
+  Variables need to be explicitly interpolated in queries with ^
+             expanding macro: Ecto.Query.where/3
+             iex:1: (file)
+             expanding macro: Ecto.Query.from/2
+             iex:1: (file)
+    (elixir) expanding macro: Kernel.|>/2
+             iex:1: (file)
+```
+
+To get around this, we use the pin operator (`^`):
+
+```elixir
+last_name = "Smith"
+Friends.Person |> Ecto.Query.where(last_name: ^last_name) |> Friends.Repo.all
+```
+
+Or:
+
+```elixir
+last_name = "Smith"
+Ecto.Query.from(p in Friends.Person, where: p.last_name == ^last_name) |> Friends.Repo.all
+```
```

Comment:
It might be good idea to explain why this is needed - the pin operator instructs the query builder to use parametrised SQL queries protecting against SQL injection